### PR TITLE
HACK week: code base tweaks (housekeeping)

### PR DESCRIPTION
**Pull request by kalessil**
Code tweaks around the codebase, encouraged by Static Code Analysis.
All CI checks shall pass.

### CODE REVIEW

File: includes/class-wc-payments-apple-pay-registration.php [hunk 1]
```diff
@@ -162,10 +162,8 @@ private function copy_and_overwrite_domain_association_file() {
 		$well_known_dir = untrailingslashit( ABSPATH ) . '/' . self::DOMAIN_ASSOCIATION_FILE_DIR;
 		$fullpath       = $well_known_dir . '/' . self::DOMAIN_ASSOCIATION_FILE_NAME;
 
-		if ( ! file_exists( $well_known_dir ) ) {
-			if ( ! @mkdir( $well_known_dir, 0755 ) ) { // @codingStandardsIgnoreLine
-				return __( 'Unable to create domain association folder to domain root.', 'woocommerce-payments' );
-			}
+		if ( ! is_dir( $well_known_dir ) && ! mkdir( $well_known_dir, 0755 ) && ! is_dir( $well_known_dir ) ) {
```
Comment (kalessil) [inline includes/class-wc-payments-apple-pay-registration.php:165]: Race conditions, when creating the directory.

File: includes/class-experimental-abtest.php [hunk 2]
```diff
@@ -101,7 +101,7 @@ protected function fetch_variation( $test_name ) {
 		}
 
 		// Make sure test name is a valid one.
-		if ( ! preg_match( '/^[A-Za-z0-9_]+$/', $test_name ) ) {
+		if ( ! preg_match( '/^\w+$/', $test_name ) ) {
```
Comment (dmvrtx) [inline includes/class-experimental-abtest.php:104]: I'm not sure if this is intended but `\w` can include Unicode characters, depending on PCRE settings. [`[:alnum:]`](https://www.php.net/manual/en/regexp.reference.character-classes.php) will be a better match for original regex.

File: includes/payment-methods/class-upe-payment-gateway.php [hunk 3]
```diff
@@ -684,7 +684,7 @@ public function get_payment_fields_js_config() {
 	 */
 	public function is_setup_intent_success_creation_redirection() {
 		return (
-			! empty( $_GET['setup_intent_client_secret'] ) & ! empty( $_GET['setup_intent'] ) & ! empty( $_GET['redirect_status'] ) && 'succeeded' === $_GET['redirect_status'] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			! empty( $_GET['setup_intent_client_secret'] ) && ! empty( $_GET['setup_intent'] ) && ! empty( $_GET['redirect_status'] ) && 'succeeded' === $_GET['redirect_status'] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Comment (dmvrtx) [inline includes/payment-methods/class-upe-payment-gateway.php:687]: This condition would be easier to read in multi-line.

**Review by dmvrtx — APPROVED** [review]
That's a lot of small but nice fixes! Great work! 🙂 

I've left a couple of comments, but nothing blocking.

File: includes/class-experimental-abtest.php (hunk 2 above)
Comment (kalessil) [reply includes/class-experimental-abtest.php]: Done in 5cd75bc13d3c4fd7c16a30232a7de65a14855925

File: includes/payment-methods/class-upe-payment-gateway.php (hunk 3 above)
Comment (kalessil) [reply includes/payment-methods/class-upe-payment-gateway.php]: Done in 5cd75bc13d3c4fd7c16a30232a7de65a14855925

**Pull request merged**